### ISSUE TITLE
Swipe broken on beta 0.117.0b0

### ISSUE THREAD

**Issue by Bergasha** [maykar/lovelace-swipe-navigation#36]
Title says it all, Updated to 0.117.0b0 and swipe component no longer works.

**Comment by maykar** [comment]
Lets see where things land in the next few versions, 0.117.0b0 is broken itself currently.

Also, "it's broken" issues are very unhelpful, try to provide as much info as possible like: Errors in your logs, errors in your dev tools console (F12 in your browser), or anything else you might find.

**Comment by maykar** [comment]
Just had a quick look. This will for sure be broken with 0.117 and I'm the one who broke it with my PRs to HA (adding compact header, etc.). Will get this fixed for 0.117 soon.

**Comment by r3mcos3** [comment]
`

Logger: frontend.js.latest.202010211
Source: components/system_log/__init__.py:193
First occurred: 7:41:23 PM (2 occurrences)
Last logged: 7:42:55 PM
http://homeassistant.local:8123/hacsfiles/lovelace-swipe-navigation/swipe-navigation.js:45:14 TypeError: tabContainer is null `

**Comment by r3mcos3** [comment]
Logger: frontend.js.latest.202010212
Source: components/system_log/__init__.py:193
First occurred: 8:30:39 AM (1 occurrences)
Last logged: 8:30:39 AM

https://ifdg1ju4i5m29bkxmsdgmbizjxuuvpuf.ui.nabu.casa/hacsfiles/lovelace-swipe-navigation/swipe-navigation.js:45:38 Uncaught TypeError: Cannot read property 'querySelectorAll' of null

**Comment by r3mcos3** [comment]
Logger: frontend.js.latest.202010212
Source: components/system_log/__init__.py:193
First occurred: 8:30:39 AM (1 occurrences)
Last logged: 8:30:39 AM

https://nabucasaurl.ui.nabu.casa/hacsfiles/lovelace-swipe-navigation/swipe-navigation.js:45:38 Uncaught TypeError: Cannot read property 'querySelectorAll' of null

**Comment by darkkatarsis** [comment]
Same here:
`http://homeassistant.local:8123/hacsfiles/lovelace-swipe-navigation/swipe-navigation.js:207:5 Uncaught TypeError: Cannot read property 'shadowRoot' of null
`

**Comment by fufar** [comment]
same at 0.117.0b6

```
Logger: frontend.js.latest.202010213
Source: components/system_log/__init__.py:193
First occurred: 20:32:01 (2 occurrences)
Last logged: 20:32:17

https://dns/hacsfiles/lovelace-swipe-navigation/swipe-navigation.js:45:38 Uncaught TypeError: Cannot read property 'querySelectorAll' of null`

Уровень: ERROR
Logger: frontend.js.latest.202010012
Source: components/system_log/__init__.py:193
First occurred: 20:31:53 (2 occurrences)
Last logged: 20:31:53

https://dns/hacsfiles/lovelace-swipe-navigation/swipe-navigation.js:207:5 Uncaught TypeError: Cannot read property 'shadowRoot' of null
https://dns/hacsfiles/kiosk-mode/kiosk-mode.js:5:36 Uncaught TypeError: Cannot read property 'shadowRoot' of null
```

**Comment by maykar** [comment]
As mentioned above, I am aware that this is broken on 0.117 (I broke it when contributing to core). There is no more need to keep posting here, thank you.

**Comment by LordNex** [comment]
Just thought I’d add my error log as well. Broke right after upgrade to 117

`Logger: frontend.js.latest.202010213
Source: components/system_log/__init__.py:193 
First occurred: October 29, 2020, 10:26:19 PM (29 occurrences) 
Last logged: 2:14:14 AM

https://mydomain.duckdns.org:8123/hacsfiles/lovelace-swipe-navigation/swipe-navigation.js:45:37 TypeError: null is not an object (evaluating 'tabContainer.querySelectorAll')
https://mydomain.duckdns.org:8123/frontend_latest/chunk.728574dc97c59021fc61.js:114:10432 TypeError: undefined is not an object (evaluating 'Object.getOwnPropertyNames(t)') `

**Comment by LordNex** [comment]
Just posted so I can get a notification. Thanks!

**Comment by Eime** [comment]
Hi guys,

found a way to fix this in the meantime.
Edit the swipe-navigation.js and change line 43 to:
`const tabContainer = appLayout.querySelector("app-header").querySelector("app-toolbar");`

@maykar Can you confirm and apply this fix?

**Comment by maykar** [comment]
That's not quite right, though it would fix it for 0.117 so good find. It should be:

```const tabContainer = appLayout.querySelector("ha-tabs");```

but we also have to be backwards compatible so:

```const tabContainer = appLayout.querySelector("paper-tabs") || appLayout.querySelector("ha-tabs");```

Updated for 0.117 compatibility and to stop errors on single tab layouts. Release version 1.2.7

**Comment by maykar** [comment]
@LordNex if you want notifications don't just post for the sake of posting, just hit subscribe on the right hand side:

![image](https://user-images.githubusercontent.com/25127328/97700726-adddd880-1a82-11eb-8c39-8fc9f95821be.png)
